### PR TITLE
Update and re-enable error retry rust test.

### PR DESCRIPTION
**Pull request by chriswldenyer**
Test to verify that retries are attempted within the nexus when IO errors are returned and when the option is enabled. The test has been modified to take account of the recent changes in error handling.

### CODE REVIEW

File: mayastor/tests/error_count_retry.rs [hunk 1]
```diff
@@ -1,139 +1,141 @@
...
+static MS: OnceCell<MayastorTest> = OnceCell::new();
 
-static YAML_CONFIG_FILE: &str = "/tmp/error_count_retry_nexus.yaml";
-#[ignore]
-#[tokio::test]
-async fn nexus_error_count_retry_test() {
-    common::truncate_file(DISKNAME1, 64 * 1024);
+static NON_ERROR_DISK: &str = "/tmp/non_error.img";
+static ERROR_DISK: &str = "/tmp/error.img";
+static NON_ERROR_BASE_BDEV: &str = "aio:////tmp/non_error.img?blk_size=512";
```
Comment (paulyoong) [inline mayastor/tests/error_count_retry.rs:23]: Is there an extra slash '/' before tmp? I usually see 3 i.e. aio:///tmp/....

File: mayastor/tests/error_count_retry.rs [hunk 2]
```diff
@@ -1,139 +1,141 @@
...
+    common::delete_file(&[NON_ERROR_DISK.to_string()]);
+    common::delete_file(&[YAML_CONFIG_FILE.to_string()]);
+}
+
+fn check_nexus_state_is(name: &str, expected_status: NexusStatus) {
+    let nexus = nexus_lookup(name).unwrap();
+    assert_eq!(nexus.status(), expected_status);
+}
+
+async fn create_nexus(name: &str, err_dev: &str, dev: &str) {
```
Comment (paulyoong) [inline mayastor/tests/error_count_retry.rs:124]: You could make this function receive a vector, then you could pass in any number of children.

**Review by paulyoong — APPROVED** [review]

File: mayastor/tests/error_count_retry.rs (hunk 1 above)
Comment (chriswldenyer) [reply mayastor/tests/error_count_retry.rs]: Now fixed, thanks

**Review by jonathan-teh — APPROVED** [review]

**Comment by chriswldenyer** [comment]
bors merge

**Comment by chriswldenyer** [comment]
bors merge

**Pull request merged**